dev/RadioButtons/RadioButtons.cpp

```diff
@@ -463,13 +500,21 @@ void RadioButtons::UpdateSelectedIndex()
     }
 }
 
-winrt::UIElement RadioButtons::ContainerFromIndex(int index)
+void RadioButtons::UpdateSelectedItem()
 {
-    if (auto const repeater = m_repeater.get())
+    if (!m_currentlySelecting)
     {
-        return repeater.TryGetElement(index);
+        if (auto const repeater = m_repeater.get())
+        {
+            if (auto const itemsSourceView = repeater.ItemsSourceView())
+            {
+                if (auto const inspectingDataSource = static_cast<InspectingDataSource*>(winrt::get_self<ItemsSourceView>(itemsSourceView)))
```

Comment:
>static_cast<InspectingDataSource*> [](start = 54, length = 34)

How do we know it's an InspectingDataSource?

Reply:
Uhm, @ranjeshj @ojhad This is what we landed on for acheiving this functionality but is ItemsSourceView guarenteed to be an InspectingDataSource? I'm guessing it is not for certain types of collections that could be passed to ItemSource... If their collection is not inspecting I think we are going to have a hard time answering index of, RadioButtons might be able to search the children of its repeater for an element who's content matches the provided data object....

Reply:
Yes. If you look at the ItemSourceView factory, what we return as the ItemsSourceView implementation is actually done by the internal type InspectingDataSource.